BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/listeners/BetterShardsListener.java

```diff
@@ -286,13 +285,16 @@ public void run() {
 				}
 				
 			});
+			event.setRespawnLocation(p.getLocation());
 			return;
 		}
 		final TeleportInfo teleportInfo = bed.getTeleportInfo();
 		if(bed.getServer() != null && bed.getServer().equals(MercuryAPI.serverName())){ //Player's bed is on the current server
 			event.setRespawnLocation(new Location(Bukkit.getWorld(teleportInfo.getWorld()), teleportInfo.getX(), teleportInfo.getY(), teleportInfo.getZ()));
 			return;
 		}
+		//respawn player where he currently is for one tick
+		event.setRespawnLocation(p.getLocation());
```

Comment:
True. We should add them to transit at this point already anyway for other reasons, just need to add something to bypass the transit check one tick later.